deploy/cdk/marble-web-kiosk-export-pipeline/lib/marble-web-kiosk-export-pipeline-stack.ts

```diff
@@ -0,0 +1,263 @@
+          actions: ['ses:SendEmail'],
+        }))
+
+        // Allow DescribeRule
+        codebuild_role.addToPolicy(new PolicyStatement({
+          resources: [ Fn.sub('arn:aws:events:${AWS::Region}:${AWS::AccountId}:rule/' + localStackName + '-*') ],
+          actions: ['events:DescribeRule', 'events:PutRule', 'events:PutTargets'],
+        }))
+
+        // Allow logging
+        codebuild_role.addToPolicy(new PolicyStatement({
+          resources: [ Fn.sub('arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/codebuild/${AWS::StackName}-*') ],
+          actions: ['logs:CreateLogGroup','logs:CreateLogStream','logs:PutLogEvents'],
+        }))
+        // Allow storing artifacts in S3 buckets
+        // Allow staging of assets in the cdk staging bucket - necessary when the application has lambda code.
+        codebuild_role.addToPolicy(new PolicyStatement({
+          resources: [
+            artifactBucket.bucketArn,
+            'arn:aws:s3:::cdktoolkit-stagingbucket-*',
+          ],
+          actions: [
+            's3:ListBucket',
+            's3:GetObject',
+            's3:PutObject',
+          ],
+        }))
+        // Allow fetching details about and updating the application stack
+        codebuild_role.addToPolicy(new PolicyStatement({
+          resources: [ Fn.sub('arn:aws:cloudformation:${AWS::Region}:${AWS::AccountId}:stack/' + localStackName + '-*/*') ],
+          actions: [
+            'cloudformation:DescribeStacks',
+            'cloudformation:DescribeStackEvents',
+            'cloudformation:DescribeChangeSet',
+            'cloudformation:CreateChangeSet',
+            'cloudformation:ExecuteChangeSet',
+            'cloudformation:DeleteChangeSet',
+            'cloudformation:DeleteStack',
+            'cloudformation:GetTemplate',
+          ],
+        }))
+        // Allow reading some details about CDKToolkit stack so we can use the CDK CLI successfully from CodeBuild.
+        codebuild_role.addToPolicy(new PolicyStatement({
+          resources: [ Fn.sub('arn:aws:cloudformation:${AWS::Region}:${AWS::AccountId}:stack/CDKToolkit/*') ],
+          actions: ['cloudformation:DescribeStacks'],
+        }))
+        // Allow modifying IAM roles related to our application
+        codebuild_role.addToPolicy(new PolicyStatement({
+          resources: [ Fn.sub('arn:aws:iam::${AWS::AccountId}:role/' + localStackName + '-*') ],
+          actions: ['iam:GetRole','iam:CreateRole','iam:DeleteRole','iam:DeleteRolePolicy','iam:AttachRolePolicy','iam:DetachRolePolicy','iam:PutRolePolicy','iam:PassRole'],
+        }))
+        // Allow full control over lambdas related to our application
+        codebuild_role.addToPolicy(new PolicyStatement({
+          resources: [ Fn.sub('arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:' + localStackName + '*') ],
+          actions: ['lambda:*'],
+        }))
+
+        return {
+          // createDns,
```

Comment:
Again, I'd take out the unused comment since it's irrelevant and unlikely to become relevant.